### PR TITLE
ENH: add interface for AFNI 3dTsmooth

### PR DESCRIPTION
**Pull request by gpiantoni**
## Summary

add interface for AFNI 3dTsmooth

## List of changes proposed in this PR (pull-request)

* add interface for AFNI 3dTsmooth
* includes almost all of the options under 3dTsmooth including:
   * Three Point Filtering Options 
   * General Linear Filtering Options 
   * Adaptive Mean Filtering option 

## Acknowledgment

- [x ] \(Mandatory\) I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

File: nipype/interfaces/afni/preprocess.py [hunk 1]
```diff
@@ -2855,6 +2855,80 @@ def _list_outputs(self):
...
+    Examples
+    ========
+
+    >>> from nipype.interfaces import afni
+    >>> from nipype.testing import  example_data
+    >>> smooth = afni.TSmooth()
+    >>> smooth.inputs.in_file = 'functional.nii'
+    >>> smooth.inputs.adaptive = 5
+    >>> smooth.cmdline
+    '3dTsmooth -prefix functional_smooth -adaptive 5 functional.nii'
```
Comment (effigies) [inline nipype/interfaces/afni/preprocess.py]:
```suggestion
    '3dTsmooth -adaptive 5 -prefix functional_smooth functional.nii'
```

File: nipype/interfaces/afni/preprocess.py [hunk 2]
```diff
@@ -2855,6 +2855,80 @@ def _list_outputs(self):
...
+        position=-1,
+        mandatory=True,
+        exists=True,
+        copyfile=False)
+    out_file = File(
+        name_template='%s_smooth',
+        desc='output file from 3dTSmooth',
+        argstr='-prefix %s',
+        name_source='in_file',
+        genfile=True)
```
Comment (effigies) [inline nipype/interfaces/afni/preprocess.py]:
```suggestion)
```

File: nipype/interfaces/afni/preprocess.py [hunk 3]
```diff
@@ -2855,6 +2855,80 @@ def _list_outputs(self):
...
+        argstr='%s',
+        position=-1,
+        mandatory=True,
+        exists=True,
+        copyfile=False)
+    out_file = File(
+        name_template='%s_smooth',
+        desc='output file from 3dTSmooth',
+        argstr='-prefix %s',
+        name_source='in_file',
```
Comment (effigies) [inline nipype/interfaces/afni/preprocess.py]:
```suggestion
        name_source='in_file')
```

**Comment by effigies** [comment]
Hi @gpiantoni, we're planning to have the next release in a little under 3 weeks, if you want to have a shot at finishing this up.

**Comment by gpiantoni** [comment]
I rebased from upstream/master and force-pushed my branch. However, one test fails (because of the change from genfile=True to genfile=None).

I tried to run `make check-before-commit` locally. This command modifies more than 40 files with very trivial changes. See output of `git diff`:
nipype

I don't think I'm supposed to commit these changes but I don't know what else to do. Sorry, I don't know how to proceed from here.

**Comment by effigies** [comment]
Hi @gpiantoni, I re-ran `make specs` on the master branch, so if you rebase again, then `make specs` should only update your test.

File: nipype/interfaces/slicer/registration/tests/test_auto_VBRAINSDemonWarp.py [hunk 4]
```diff
@@ -1,118 +0,0 @@
-# AUTO-GENERATED by tools/checkspecs.py - DO NOT EDIT
```
Comment (effigies) [inline nipype/interfaces/slicer/registration/tests/test_auto_VBRAINSDemonWarp.py]: I don't think this file should have been removed.

Comment (gpiantoni) [reply nipype/interfaces/slicer/registration/tests/test_auto_VBRAINSDemonWarp.py]: Sorry, should be fixed now

**Comment by effigies** [comment]
Thanks for your patience, @gpiantoni. In it goes.

**Pull request merged**